test/decorator-request.test.js

```diff
@@ -0,0 +1,49 @@
+import { test } from 'tap'
+import Fastify from 'fastify'
+import supertest from 'supertest'
+
+import slowDownPlugin from '../index.js'
+import { DEFAULT_OPTIONS } from '../lib/constants.js'
+import { slowDownAPI } from './helpers.js'
+import { convertToMs } from '../lib/helpers.js'
+
+test('should contain the slowDown request decorator', async t => {
+  t.test('without delay property', async t => {
+    const fastify = Fastify()
+    await fastify.register(slowDownPlugin)
+    t.teardown(() => fastify.close())
+
+    fastify.get('/', async req => req.slowDown)
+
+    const responseBody = await (await fastify.inject('/')).json()
+
+    t.has(responseBody, {
+      limit: DEFAULT_OPTIONS.delayAfter,
+      current: 1,
+      remaining: DEFAULT_OPTIONS.delayAfter - 1,
+      delay: undefined
+    })
+  })
+
+  t.test('with delay property', async t => {
+    const fastify = Fastify()
+    await fastify.register(slowDownPlugin)
+    t.teardown(() => fastify.close())
+
+    fastify.get('/', async req => req.slowDown)
+
+    await fastify.ready()
+
+    await slowDownAPI(DEFAULT_OPTIONS.delayAfter, () =>
+      supertest(fastify.server).get('/')
```

Comment:
why did you have to use supertest here?

Reply:
https://github.com/nearform/fastify-slow-down/pull/13#issuecomment-1254782630

`fastify.inject` is not waiting for the timeout inside promise and it returns an empty response: 
https://github.com/nearform/fastify-slow-down/blob/c070660e4d1f992b2a002a144bd82c231efe3e9d/index.js#L44
https://nearform.slack.com/archives/D03VB4J3EKG/p1663582314738489

Reply:
I still don't understand. we're never testing that scenario here